### PR TITLE
#RI-3571,3564,3560 - recommendations

### PR DESCRIPTION
**Pull request by AmirAllayarovSofteq**

### CODE REVIEW

**Review by arthosofteq — CHANGES_REQUESTED** [review]
Well done! I like this! 
Please, resolve small comments and you could process with tests. I'll review final PR with tests next time

File: redisinsight/api/src/modules/recommendation/recommendation.service.ts [hunk 1]
```diff
@@ -37,6 +37,9 @@ export class RecommendationService {
       await this.recommendationProvider.determineLuaScriptRecommendation(client),
       await this.recommendationProvider.determineBigHashesRecommendation(keys),
       await this.recommendationProvider.determineBigTotalRecommendation(total),
+      await this.recommendationProvider.determineLogicalDatabasesRecommendation(client),
+      await this.recommendationProvider.determineCombineSmallStringsToHashesRecommendation(keys),
+      await this.recommendationProvider.determineIncreaseSetMaxIntsetEntriesRecommendation(client, keys),
     ]);
```
[the remark this answers is not in the document]
Comment (AmirAllayarovSofteq) [reply redisinsight/api/src/modules/recommendation/recommendation.service.ts:43]: Good point, thx

**Review by arthosofteq — APPROVED** [review]
lgtm
Also, please, use constants for recommendation names. You can implement this later with other recommendations next time

File: redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts [hunk 2]
```diff
@@ -51,4 +57,103 @@ export class RecommendationProvider {
...
+  }
+
+  /**
+   * Check combine small strings to hashes recommendation
+   * @param keys
+   */
+  async determineCombineSmallStringsToHashesRecommendation(
+    keys: Key[],
+  ): Promise<Recommendation> {
+    const smallString = keys.some((key) => key.type === RedisDataType.String && key.memory < maxStringMemory);
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts:95]: try, catch?

File: redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts [hunk 3]
```diff
@@ -51,4 +57,103 @@ export class RecommendationProvider {
...
+
+  /**
+   * Check compress hash field names recommendation
+   * @param keys
+   */
+  async determineCompressHashFieldNamesRecommendation(
+    keys: Key[],
+  ): Promise<Recommendation> {
+    const bigHash = keys.some((key) => key.type === RedisDataType.Hash && key.length > maxCompressHashLength);
+    return bigHash ? { name: 'compressHashFieldNames' } : null;
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts:157]: try, catch?

File: redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts [hunk 4]
```diff
@@ -51,4 +57,103 @@ export class RecommendationProvider {
...
+        ) as string,
+      );
+      const keyspace = get(info, 'keyspace', {});
+      const databasesWithKeys = Object.values(keyspace).filter((db) => {
+        const { keys } = convertBulkStringsToObject(db as string, ',', '=');
+        return keys > 0;
+      });
+      return databasesWithKeys.length > 1 ? { name: 'avoidLogicalDatabases' } : null;
+    } catch (err) {
+      return null;
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts:84]: also you could add some logs on error to track such errors

**Review by arthosofteq — APPROVED** [review]

File: redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts (hunk 2 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts:95]: Why we need try catch here ?

File: redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts (hunk 4 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts:84]: Will add in next recommendation

**Pull request merged**